### PR TITLE
Add a feature to `swap_year_for_time`

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR introduces the "inverse" of the `swap_time_for_year()` method. This should facilitate conversion from and to proper `datetime` timeseries and the "subannual" column currently used in the IIASA database infrastructure.

As shown in the tests, if you have an IamDataFrame with datetime format, then

```python
df.swap_time_for_year(subannual=True).swap_year_for_time()
```
will yield an identical object.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
@loeffko @sebastianzwickl @erikfilias, this may be helpful for your ongoing work in the model linkage for openENTRANCE.

File: pyam/core.py [hunk 1]
```diff
@@ -613,25 +612,58 @@ def interpolate(self, time, inplace=False, **kwargs):
...
             "time" is not a column of `self.data`
+
+        See Also
+        --------
+        swap_year_for_time
         """
         return swap_time_for_year(self, subannual=subannual, inplace=inplace)
 
+    def swap_year_for_time(self, inplace=False):
+        """Convert the `year` and `subannual` dimensions to `time` (as datetime).
```
Comment (znicholls) [inline pyam/core.py:644]: Worth adding that the conversion is done using `dateutil.parser.parse` on the combination of year and subannual?

**Review by znicholls — COMMENTED** [review]
My one quick thought. Lots of edge cases could be covered here, depends how widely this feature is expected to be used.

File: pyam/core.py (hunk 1 above)
Comment (danielhuppmann) [reply pyam/core.py:645]: Thanks, great suggestion - added this to the docstring including intersphinx mapping...

**Comment by danielhuppmann** [comment]
> My one quick thought. Lots of edge cases could be covered here, depends how widely this feature is expected to be used.

The unit tests already cover five basic cases, including the one we are using in the openENTRANCE and ECEMF projects - this is the one where I see this function actually being used.

https://github.com/danielhuppmann/pyam/blob/5adccd81a8416aac3ed185b7bd33a7bb7c0cb515/tests/test_time.py#L55

I don't see any other use cases on the horizon (like using a different column name than "subannual" for the non-year parts, or using several columns) - but no reason why these two "swap" functions couldn't be extended in that direction with keyword arguments...

**Comment by danielhuppmann** [comment]
Merging as there were no objections or further comments... Thanks @znicholls for the suggestion to improve the docs!
